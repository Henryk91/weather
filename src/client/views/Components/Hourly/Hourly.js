import React, { Component } from 'react'
import ReactAnimatedWeather from 'react-animated-weather';
import { getTime, convertIconName } from '../../Helpers/helper-functions';
// import './style.css';

export default class Hourly extends Component {
    constructor(props) {
        super(props);
    }

    render() {
    let hourLine = null;
    var weatherData = this.props.weatherData;
    if (weatherData) {
        hourLine = weatherData.hourly.data.map((val, i) => {
            let date = new Date(val.time * 1000).toDateString()
                date = date.substring(0, (date.length -4))
            return (
                <div key={val.time} className="weatherLine">
                    <p>{getTime(val.time)} {date}</p>
                    <p>{val.temperature}&#176;</p>
                    <p>Wind: {val.windSpeed}</p>
                    <p>Persip: {Math.round(val.precipProbability*100)}%</p>
                    <ReactAnimatedWeather
                        icon={convertIconName(val.icon)}
                        color={'#FEBF34'}
                        size={40}
                        animate={true}
                    />
                </div>
            )
        })
    }
    return (
      <div>
        <h1>Hourly Forcast</h1>
        {hourLine}
      </div>
    )
  }
}
