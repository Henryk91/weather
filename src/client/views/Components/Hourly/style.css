.weatherLine{
    margin: 2px;
    display:grid;
    grid-template-columns:repeat(5, 1fr);
    grid-auto-rows: 1fr;
    background: #007998;
  }