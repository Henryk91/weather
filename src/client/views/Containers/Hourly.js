import React, { Component } from 'react'
import ReactAnimatedWeather from 'react-animated-weather';

export default class Hourly extends Component {
    constructor(props) {
        super(props);
        this.state = {
            weatherData: this.props.weatherData,

        };
        this.getTime = this.getTime.bind(this)
        this.convertIcon = this.convertIcon.bind(this)
    }

    convertIcon(icon) {
        return icon.toUpperCase().replace(/-/g, "_")
    }

    getTime(time) {
        var date = new Date(time * 1000)
        var dateTime = date.toTimeString()
        return dateTime.substring(0, dateTime.lastIndexOf(":"))
    }

    render() {
    let hourLine = null;
    if (this.state.weatherData) {
        hourLine = this.state.weatherData.hourly.data.map((val, i) => {
            let date = new Date(val.time * 1000).toDateString()
                date = date.substring(0, (date.length -4))
            return (
                <div key={val.time} className="weatherLine">
                    <p>{this.getTime(val.time)} {date}</p>
                    <p>{val.temperature}&#176;</p>
                    <p>Wind: {val.windSpeed}</p>
                    <p>Persip: {Math.round(val.precipProbability*100)}%</p>
                    <ReactAnimatedWeather
                        icon={this.convertIcon(val.icon)}
                        color={'#FEBF34'}
                        size={40}
                        animate={true}
                    />
                </div>
            )
        })
    }
    return (
      <div>
        <h1>Hourly Forcast</h1>
        {hourLine}
      </div>
    )
  }
}
