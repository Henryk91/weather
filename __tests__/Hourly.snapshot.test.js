
import Hourly from '../src/client/views/Components/Hourly/Hourly';
import renderer from 'react-test-renderer';
import React from 'react';

import { weatherData } from "../testData/weatherData.sample"

test('Hourly component renders all hours', () => {

  const rendered = renderer.create(
    <Hourly weatherData={weatherData()}/>
  );
  expect(rendered.toJSON()).toMatchSnapshot();
});