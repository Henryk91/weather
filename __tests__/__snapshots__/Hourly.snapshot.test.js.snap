// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hourly component renders all hours 1`] = `
<div>
  <h1>
    Hourly Forcast
  </h1>
  <div
    className="weatherLine"
  >
    <p>
      19:00
       
      Fri Mar 08 
    </p>
    <p>
      24.73
      °
    </p>
    <p>
      Wind: 
      3.33
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      20:00
       
      Fri Mar 08 
    </p>
    <p>
      23.02
      °
    </p>
    <p>
      Wind: 
      2.97
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      21:00
       
      Fri Mar 08 
    </p>
    <p>
      22.15
      °
    </p>
    <p>
      Wind: 
      2.65
    </p>
    <p>
      Persip: 
      4
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      22:00
       
      Fri Mar 08 
    </p>
    <p>
      21.88
      °
    </p>
    <p>
      Wind: 
      1.82
    </p>
    <p>
      Persip: 
      5
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      23:00
       
      Fri Mar 08 
    </p>
    <p>
      21.61
      °
    </p>
    <p>
      Wind: 
      1.25
    </p>
    <p>
      Persip: 
      8
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      00:00
       
      Sat Mar 09 
    </p>
    <p>
      21.22
      °
    </p>
    <p>
      Wind: 
      1.03
    </p>
    <p>
      Persip: 
      8
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      01:00
       
      Sat Mar 09 
    </p>
    <p>
      20.96
      °
    </p>
    <p>
      Wind: 
      0.77
    </p>
    <p>
      Persip: 
      8
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      02:00
       
      Sat Mar 09 
    </p>
    <p>
      20.58
      °
    </p>
    <p>
      Wind: 
      0.6
    </p>
    <p>
      Persip: 
      8
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      03:00
       
      Sat Mar 09 
    </p>
    <p>
      19.57
      °
    </p>
    <p>
      Wind: 
      0.5
    </p>
    <p>
      Persip: 
      7
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      04:00
       
      Sat Mar 09 
    </p>
    <p>
      18.41
      °
    </p>
    <p>
      Wind: 
      0.53
    </p>
    <p>
      Persip: 
      4
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      05:00
       
      Sat Mar 09 
    </p>
    <p>
      18.04
      °
    </p>
    <p>
      Wind: 
      0.66
    </p>
    <p>
      Persip: 
      4
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      06:00
       
      Sat Mar 09 
    </p>
    <p>
      18.96
      °
    </p>
    <p>
      Wind: 
      0.59
    </p>
    <p>
      Persip: 
      3
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      07:00
       
      Sat Mar 09 
    </p>
    <p>
      20.4
      °
    </p>
    <p>
      Wind: 
      0.65
    </p>
    <p>
      Persip: 
      3
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      08:00
       
      Sat Mar 09 
    </p>
    <p>
      22.16
      °
    </p>
    <p>
      Wind: 
      0.84
    </p>
    <p>
      Persip: 
      3
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      09:00
       
      Sat Mar 09 
    </p>
    <p>
      24.39
      °
    </p>
    <p>
      Wind: 
      1.09
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      10:00
       
      Sat Mar 09 
    </p>
    <p>
      26.61
      °
    </p>
    <p>
      Wind: 
      1.44
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      11:00
       
      Sat Mar 09 
    </p>
    <p>
      28.11
      °
    </p>
    <p>
      Wind: 
      1.78
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      12:00
       
      Sat Mar 09 
    </p>
    <p>
      29.59
      °
    </p>
    <p>
      Wind: 
      2.11
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      13:00
       
      Sat Mar 09 
    </p>
    <p>
      30.61
      °
    </p>
    <p>
      Wind: 
      2.42
    </p>
    <p>
      Persip: 
      1
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      14:00
       
      Sat Mar 09 
    </p>
    <p>
      31.14
      °
    </p>
    <p>
      Wind: 
      2.61
    </p>
    <p>
      Persip: 
      1
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      15:00
       
      Sat Mar 09 
    </p>
    <p>
      31.06
      °
    </p>
    <p>
      Wind: 
      2.57
    </p>
    <p>
      Persip: 
      1
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      16:00
       
      Sat Mar 09 
    </p>
    <p>
      30.58
      °
    </p>
    <p>
      Wind: 
      2.41
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      17:00
       
      Sat Mar 09 
    </p>
    <p>
      29.66
      °
    </p>
    <p>
      Wind: 
      2.17
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      18:00
       
      Sat Mar 09 
    </p>
    <p>
      28.03
      °
    </p>
    <p>
      Wind: 
      1.9
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      19:00
       
      Sat Mar 09 
    </p>
    <p>
      25.96
      °
    </p>
    <p>
      Wind: 
      1.59
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      20:00
       
      Sat Mar 09 
    </p>
    <p>
      24.29
      °
    </p>
    <p>
      Wind: 
      1.35
    </p>
    <p>
      Persip: 
      0
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      21:00
       
      Sat Mar 09 
    </p>
    <p>
      23.34
      °
    </p>
    <p>
      Wind: 
      1.16
    </p>
    <p>
      Persip: 
      2
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      22:00
       
      Sat Mar 09 
    </p>
    <p>
      22.72
      °
    </p>
    <p>
      Wind: 
      1.01
    </p>
    <p>
      Persip: 
      3
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      23:00
       
      Sat Mar 09 
    </p>
    <p>
      22.2
      °
    </p>
    <p>
      Wind: 
      0.9
    </p>
    <p>
      Persip: 
      5
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      00:00
       
      Sun Mar 10 
    </p>
    <p>
      21.61
      °
    </p>
    <p>
      Wind: 
      0.84
    </p>
    <p>
      Persip: 
      6
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      01:00
       
      Sun Mar 10 
    </p>
    <p>
      21.24
      °
    </p>
    <p>
      Wind: 
      0.84
    </p>
    <p>
      Persip: 
      9
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      02:00
       
      Sun Mar 10 
    </p>
    <p>
      21.04
      °
    </p>
    <p>
      Wind: 
      0.85
    </p>
    <p>
      Persip: 
      12
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      03:00
       
      Sun Mar 10 
    </p>
    <p>
      21
      °
    </p>
    <p>
      Wind: 
      0.87
    </p>
    <p>
      Persip: 
      13
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      04:00
       
      Sun Mar 10 
    </p>
    <p>
      21.07
      °
    </p>
    <p>
      Wind: 
      0.9
    </p>
    <p>
      Persip: 
      13
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      05:00
       
      Sun Mar 10 
    </p>
    <p>
      21.27
      °
    </p>
    <p>
      Wind: 
      0.96
    </p>
    <p>
      Persip: 
      15
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      06:00
       
      Sun Mar 10 
    </p>
    <p>
      21.41
      °
    </p>
    <p>
      Wind: 
      1.05
    </p>
    <p>
      Persip: 
      16
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      07:00
       
      Sun Mar 10 
    </p>
    <p>
      21.48
      °
    </p>
    <p>
      Wind: 
      1.17
    </p>
    <p>
      Persip: 
      20
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      08:00
       
      Sun Mar 10 
    </p>
    <p>
      22.33
      °
    </p>
    <p>
      Wind: 
      1.27
    </p>
    <p>
      Persip: 
      19
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      09:00
       
      Sun Mar 10 
    </p>
    <p>
      24.34
      °
    </p>
    <p>
      Wind: 
      1.35
    </p>
    <p>
      Persip: 
      13
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      10:00
       
      Sun Mar 10 
    </p>
    <p>
      26.92
      °
    </p>
    <p>
      Wind: 
      1.43
    </p>
    <p>
      Persip: 
      7
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      11:00
       
      Sun Mar 10 
    </p>
    <p>
      28.35
      °
    </p>
    <p>
      Wind: 
      1.54
    </p>
    <p>
      Persip: 
      7
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      12:00
       
      Sun Mar 10 
    </p>
    <p>
      29.05
      °
    </p>
    <p>
      Wind: 
      1.75
    </p>
    <p>
      Persip: 
      13
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      13:00
       
      Sun Mar 10 
    </p>
    <p>
      28.87
      °
    </p>
    <p>
      Wind: 
      1.98
    </p>
    <p>
      Persip: 
      16
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      14:00
       
      Sun Mar 10 
    </p>
    <p>
      28.1
      °
    </p>
    <p>
      Wind: 
      2.06
    </p>
    <p>
      Persip: 
      22
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      15:00
       
      Sun Mar 10 
    </p>
    <p>
      27.12
      °
    </p>
    <p>
      Wind: 
      1.91
    </p>
    <p>
      Persip: 
      23
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      16:00
       
      Sun Mar 10 
    </p>
    <p>
      26.46
      °
    </p>
    <p>
      Wind: 
      1.63
    </p>
    <p>
      Persip: 
      25
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      17:00
       
      Sun Mar 10 
    </p>
    <p>
      25.63
      °
    </p>
    <p>
      Wind: 
      1.38
    </p>
    <p>
      Persip: 
      27
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      18:00
       
      Sun Mar 10 
    </p>
    <p>
      24.81
      °
    </p>
    <p>
      Wind: 
      1.19
    </p>
    <p>
      Persip: 
      30
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
  <div
    className="weatherLine"
  >
    <p>
      19:00
       
      Sun Mar 10 
    </p>
    <p>
      23.7
      °
    </p>
    <p>
      Wind: 
      1.04
    </p>
    <p>
      Persip: 
      34
      %
    </p>
    <canvas
      height={40}
      width={40}
    />
  </div>
</div>
`;
